File: King/src/org/uva/sea/ql/ast/expr/VarExpr.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.VarIdentifier;
import org.uva.sea.ql.ast.expr.type.Type;
import org.uva.sea.ql.ast.visitors.QLNodeVisitor;

public class VarExpr extends Expr {
	private final VarIdentifier identifier;

	public VarExpr(VarIdentifier identifier) {
		this.identifier = identifier;
	}

	public VarIdentifier getIdentifier() {
		return identifier;
	}
	
	public String getIdentifierName() {
		return identifier.getName();
	}

	public Type type() {
		return identifier.getType();
	}

	public boolean equals(VarExpr varExpr) {
		return identifier.getName().equals(varExpr.getIdentifier().getName());
	}

	@Override
	public String toString() {
		return "[ " + identifier.toString() + " ]";
	}

	@Override
	public <T> T accept(QLNodeVisitor<T> visitor) {
		return visitor.visit(this);

	}

}

File: Edward Poot/src/QL/ast/expr/varexpr.go
package expr

import (
	"ql/ast/vari"
	"ql/ast/visit"
)

type VarExpr struct {
	Identifier vari.VarId
}

// TODO implement
func (v VarExpr) Eval(symTable interface{}) interface{} {
	return false
}

func (v VarExpr) Accept(vis visit.Visitor) interface{} {
	return vis.Visit(v)
}
